java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -381,10 +495,19 @@ private static Consumer createStructConsumer(Schema schema, String name, AvroToA
     }
 
     Consumer[] delegates = new Consumer[schema.getFields().size()];
+    int vectorIndex = 0;
     for (int i = 0; i < schema.getFields().size(); i++) {
-      Schema childSchema = schema.getFields().get(i).schema();
-      Consumer delegate = createConsumer(childSchema, childSchema.getName(),
-          config, structVector.getChildrenFromFields().get(i));
+      Schema.Field childField = schema.getFields().get(i);
+      Consumer delegate;
+      // use full name to distinguish fields have same names between parent and child fields.
+      final String fullChildName = String.format("%s.%s", name, childField.name());
```

Comment:
doesn't this work just one level deep?

Reply:
No, the name could be format like "%s.%s", the fullChildName could be passed on to the next level.

Reply:
please add a test for this.

Reply:
https://github.com/apache/arrow/pull/5482/files#diff-903c589a8f5633f9720e44dee3efdc2bR158
This test already covered this case.